### ISSUE TITLE
TODO, by_s_m.38 - error description?

### ISSUE THREAD

**Issue by DougCollinge** [boshek/limnotools#14]
The TODO suggests replacing an error signal return of -99 with a crash but I'm not sure that this is the right thing to do. I don't like the way it works now because -99 has no intrinsic meaning, it just shows up on the report and you the human reader are just supposed to intuit that "something went wrong" and ignore the rest of the report. That is seriously bogus, IMHO. What I think should actually happen here is that the report should just be replaced with a detailed explanation of why the report could not be produced. In order to accomplish this by_s_m needs to be modified to allow an exit that conveys the error message, whatever it is. Perhaps it should always return a list containing an error code, error explanation, and data, if no errors.

**Comment by DougCollinge** [comment]
In fact, by_s_m has another error situation not handled at line 44. This problem should be rolled into the same fix, IMHO.

**Comment by DougCollinge** [comment]
by_s_m3 has the same issues, of course.

**Comment by boshek** [comment]
Absolutely. -99 isn't helpful at all. At least something that gives the user an idea.

**Comment by DougCollinge** [comment]
It seems that there is a package that implements try/throw/catch exceptions, which would do this job in a standard way. I'm reluctant to add a heavy dependency, however, just for this little problem. Is there something like that in that "master" roll-up package we already depend on? Sorry, can't remember what it is called.

**Comment by DougCollinge** [comment]
Ok, tentatively done for by_s_m. You just use stop() to throw an error and tryCatch can catch the error. I don't know how to document this possibility so I haven't done it.

**Comment by boshek** [comment]
A stop and then maybe an informative error message?
